tsdoc/src/beta/DeclarationReference.grammarkdown

```diff
@@ -34,7 +34,10 @@ MeaningKeyword: one of
   `member`                                      // SymbolFlags.ClassMember
   `event`                                       //
   `enummember`                                  // SymbolFlags.EnumMember
-  `signature`                                   // SymbolFlags.Signature
+  `call`                                        // SymbolFlags.Signature (for __call)
+  `new`                                         // SymbolFlags.Signature (for __new)
+  `index`                                       // SymbolFlags.Signature (for __index)
+  `signature`                                   // SymbolFlags.Signature (deprecated)
   `type`                                        // Any complex type

```

Comment:
@rbuckton In https://github.com/microsoft/web-build-tools/pull/1406#discussion_r305665716 I am proposing the following list of meanings:

- `call`
- `class`
- `constructor`
- `enum`
- `function`
- `imember` (covers `MethodSignature` and `PropertySignature`)
- `index`
- `interface`
- `member` (covers `EnumMember`, `Method`, and `Property`)
- `namespace`
- `new`
- `type`  (replaces `typealias` because I believe it's more intuitive to use TypeScript keywords than compiler internals)
- `var`

What do you think?

Reply:
- `imember` - unnecessary, `member` works fine for these.
- `var` - I suppose its shorter than `variable`
- `type` - There is already a `type` and it has a different purpose than `typealias`. "type" is also a very general term, while a "type alias" is what a `type` keyword produces. I disagree that we should use the keyword over the semantic meaning here.

Reply:
> * `imember` - unnecessary, `member` works fine for these.

This seems like a counterexample:

```ts
interface X {
  y: number; // X#y:member
}

class X {
  public y: number;  // X#y:member
}
```

How would you handle it?

Reply:
As far as TypeScript is concerned, they are the same member. If you try to change the type of one to `"string"`, you get a "duplicate identifier" error because they collide.

Reply:
case in point:
```ts
interface X {
  y: number;
}
interface X {
  y: number; // change to `string` to get the same "duplicate identifier" error
}
```

Reply:
Here's a counterproposal:
- `call` for `SyntaxKind.Signature` with name `__call`
- `new` for `SyntaxKind.Signature` with name `__new`
- `index` for `SyntaxKind.Signature` with name `__index`
- `variable` becomes `var`
- `type` becomes `complex`
- `typealias` becomes `type`
- `enummember` becomes `member`
- `member` satisfies the following meanings:
  - Method
  - MethodSignature
  - Property
  - PropertySignature
  - GetAccessor
  - SetAccessor
  - EnumMember

There *is* a possible conflict in meanings for `member`, but it seems trivial as TypeScript still treats them as the same member:

```ts
class C {
  get x(): number;
}
interface C {
  x: number;
}
```

Reply:
Think about it more, the `class`+`interface` merge is fairly bizarre.  Nobody can implement the interface without also implementing the class members.  It's a bit of a puzzle how the documentation system should even represent that.   I agree that `imember` won't be needed or meaningful in practice.

👍 I'm good with this proposal.